[Tung Yan Zuo] iP

Darren-Tung opened the pull request:
# iP Project Pull Request: Code Refactoring of Chatbot 😉
This pull request includes significant improvements to the code structure and documentation. Below are the details of the changes made:

## Key Changes Implemented
1. Created a chatbot named 'Tuesday'.
2. Added Javadoc comments across multiple classes.
3. Resolved Checkstyle issues related to import ordering, indentation, and line length.

## Created classes
There are many types of classes. Below is a list of classes and what they are under. 

### The chatbot itself
- Tuesday

### Command Types
- Command
- AddCommand
- MarkCommand
- ListCommand
- ExitCommand
- FindCommand
- DeleteCommand
- Parser

### Interface Types
- DialogBox
- Launcher
- Main
- MainWindow

### Task Types
- Task
- Deadline
- ToDo
- Event

## Extra
I **love** this [song](https://www.youtube.com/watch?v=Oextk-If8HQ) 
> Somewhere only we know
Do you like this song too?
- [x] Yes
- [ ] No

Hello world in Java
```
class HelloWorld {
    public static void main(String[] args) {
        System.out.println("Hello, World!"); 
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Darren-Tung committed:
Update Duke.java

Darren-Tung committed:
Week 2 Level 0

Added the Tuesday.java file and added the dialogues

Darren-Tung committed:
Update Tuesday.java

Changed the messages to methods and added list and blah

Darren-Tung committed:
Week 2 Level 1

Updated the code to echo commands entered by the user, and exits when the user types the command bye

Darren-Tung committed:
Merge pull request #1 from Darren-Tung/working

Week 2 Level 1 done

Darren-Tung committed:
Week 2 Level 2

Level 2. Add, List
Add the ability to store whatever text entered by the user and display them back to the user when requested.

Darren-Tung committed:
Merge pull request #2 from Darren-Tung/working

Week 2 Level 2

Darren-Tung committed:
Week 2 Level 3

Added a Task class and add the ability to mark tasks as done. Optionally, add the ability to change the status back to not done.

Darren-Tung committed:
Merge pull request #3 from Darren-Tung/working

Week 2 Level 3

Darren-Tung committed:
Skeleton for week 2 level 4

Added 3 new classes, inherited from task.
Changed the code to work for the new format

Darren-Tung committed:
Week 2 Level 4

Add support for tracking three types of tasks:

ToDos: tasks without any date/time attached to it e.g., visit new theme park
Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
Events: tasks that start at a specific date/time and ends at a specific date/time
e.g., (a) team project meeting 2/10/2019 2-4pm (b) orientation week 4/10/2019 to 11/10/2019

Darren-Tung committed:
Merge pull request #4 from Darren-Tung/working

Week 2 Level 4

Darren-Tung committed:
Week 2 A-TextUiTesting

Updated the Text UI Testing files

Darren-Tung committed:
Week 2 Level 5

Used error checking

Darren-Tung committed:
Week 2 Level 6

Added support for deleting tasks from the list.

Darren-Tung committed:
Merge pull request #5 from Darren-Tung/working

Doing level 5 and 6

Darren-Tung committed:
Work on Level 7

Darren-Tung committed:
Finish Level 7

Darren-Tung committed:
Update deadline function to understand date & time

Only for deadline function

Darren-Tung committed:
Finish Event function to read date & time

Darren-Tung committed:
Complete skeleton of the code

Darren-Tung committed:
Update all classes to A-MoreOOP standard

Darren-Tung committed:
Merge pull request #6 from Darren-Tung/branch-Level-7

Branch level 7

Darren-Tung committed:
Merge branch 'master' into branch-Level-8

Darren-Tung committed:
Merge pull request #7 from Darren-Tung/branch-Level-8

Branch level 8

Darren-Tung committed:
Merge branch 'master' into A-MoreOOP

Darren-Tung committed:
Merge pull request #8 from Darren-Tung/A-MoreOOP

Finish A-MoreOOP

Darren-Tung committed:
Fix minor bugs

Darren-Tung committed:
Move files to new folders

Darren-Tung committed:
Merge pull request #9 from Darren-Tung/A-Packages

Move files to new folders

Darren-Tung committed:
Merge branch 'add-gradle-support'

Darren-Tung committed:
Create test cases for Tuesday

Darren-Tung committed:
A-Jar

Darren-Tung committed:
Update the javadoc

Darren-Tung committed:
Update coding standard

Darren-Tung committed:
Add find function

Darren-Tung committed:
Merge pull request #10 from Darren-Tung/branch-A-JavaDoc

Update the javadoc

Darren-Tung committed:
Merge branch 'master' into branch-A-CodingStandard

Darren-Tung committed:
Merge pull request #11 from Darren-Tung/branch-A-CodingStandard

Update coding standard

Darren-Tung committed:
Merge branch 'master' into branch-Level-9

Darren-Tung committed:
Merge pull request #12 from Darren-Tung/branch-Level-9

Add find function

Darren-Tung committed:
Fix bugs

Ponharish commented on src/main/java/Tuesday/Tuesday.java:
```diff
@@ -0,0 +1,53 @@
+     */
+    public Tuesday(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
```
Have you considered the situation where the file or folder doesn't exist?

Darren-Tung replied:
Thanks for the help. I forgot to consider that.

Ponharish commented on src/main/java/Tuesday/command/Command.java:
```diff
@@ -0,0 +1,17 @@
+package Tuesday.command;
+
+import Tuesday.util.Storage;
```
You may want to consider ordering your import statement to be in alphabetical order

Ponharish commented on src/main/java/Tuesday/command/DeleteCommand.java:
```diff
@@ -0,0 +1,83 @@
+import Tuesday.task.Task;
+import Tuesday.util.Ui;
+
+import java.io.*;
```
Perhaps you may want to list out only the specific packages you would be using?

Darren-Tung replied:
I did not realise this. Thank you

Ponharish commented on src/main/java/Tuesday/command/DeleteCommand.java:
```diff
@@ -0,0 +1,83 @@
+                    i++;
+                }
+                String everything = sb.toString();
+                //System.out.println(everything);
```
You may consider deleting this line

Ponharish commented on src/main/java/Tuesday/command/Parser.java:
```diff
@@ -0,0 +1,41 @@
+    }
+
+    /**
+     * Takes in the command and allocates the corresponding method
```
I like the Javadoc comments given for all methods 👍

Ponharish reviewed (commented):
Overall, I found your code to be well written and easy to read. I have read the first few files and so far, it just requires some minor changes mostly, coding standard violations. 

Good work 👍

rithiklenin commented on src/main/java/Tuesday/command/AddCommand.java:
```diff
@@ -0,0 +1,67 @@
+     * @return false and do not exit
+     */
+    @Override
+    public boolean isExit() {
```
Might consider changing the method name to hasExited() if it refers to checking if program has exited

Darren-Tung replied:
Thank you for the suggestion

rithiklenin commented on src/main/java/Tuesday/command/DeleteCommand.java:
```diff
@@ -0,0 +1,83 @@
+     * Deletes the task from the data file
+     */
+    public void deleteDataFromFile() {
+        int i = 0;
```
You can consider changing i to count, but it is fine either ways

rithiklenin commented on src/main/java/Tuesday/command/DeleteCommand.java:
```diff
@@ -0,0 +1,83 @@
+                }
+                String everything = sb.toString();
+                //System.out.println(everything);
+                FileWriter wr = new FileWriter(new File("src/main/data/tuesday.txt"), false);
```
Can consider renaming wr to fw

rithiklenin commented on src/main/java/Tuesday/command/MarkCommand.java:
```diff
@@ -0,0 +1,84 @@
+import Tuesday.task.Task;
+import Tuesday.util.Ui;
+
+import java.io.*;
```
Should list out the imported classes or methods explicitly

rithiklenin commented on src/main/java/Tuesday/command/MarkCommand.java:
```diff
@@ -0,0 +1,84 @@
+public class MarkCommand extends Command {
+    // variable
+    private int taskIndex;
+    private boolean isMarked;
```
Nice! The boolean variable name uses the right verb

rithiklenin reviewed (commented):
Overall welldone! Minor changes to look at.

Darren-Tung committed:
Update the styling of the code based on Stylecheck

- Change folder name 'Tuesday' to 'tuesday'
- Used the stylecheck code to refactor my code

Darren-Tung committed:
Update gradle for JavaFX

Darren-Tung committed:
Update to JavaFX Tutorial 1 level

Darren-Tung committed:
Update to JavaFX Tutorial 2

Darren-Tung committed:
Update JavaFX Tutorial 3 level

Darren-Tung committed:
Update JavaFX Tutorial 4 level

Darren-Tung committed:
Update JavaFX Tutorial 5 level

Darren-Tung committed:
Update code to work with the tutorial

Darren-Tung committed:
Add a deadline test under CommandTest

Previously, the test to test the deadline function did not exist

There is a need to test the deadline function as the code is everchanging.

Things changed:
- Add commandExecuteCheckDatafileForDeadline()
- Fix handleByMsg() logic

Darren-Tung committed:
Update gradle to run assert and add assert to command methods

Added some additional assert functions

Darren-Tung committed:
Add assert to Task commands

Darren-Tung committed:
Add assert to the functions

Darren-Tung committed:
Remove arrow heads

The aim is to remove arrow heads from the methods.

Things changed:
- Removed if statement from deadline.java
- Removed if statement from event.java
- Removed if statement from todo.java
- Added a helper function in storage.java
- Added a helper function in ui.java

Overall, this is to reduce the number of arrow heads

Darren-Tung committed:
Update using SLAP

Darren-Tung committed:
Merge branch 'master' into branch-A-CodeQuality

Darren-Tung committed:
Merge pull request #13 from Darren-Tung/branch-A-CodeQuality

Branch Code Quality branch

Darren-Tung committed:
Update C-Priority

Darren-Tung committed:
Fix bugs and create help function

Update list:
 - Edit command: update the data file
 - Help command: gives a list of available functions

Edit command needs to update data file. It is expected for the edit command to update the data file as well.
Tuesday needs a function to display a list of available functions.

Let's,
* add a editDaraFromFile() function
* add a Help class
* update the code to accommodate to the changes

Darren-Tung committed:
Update Tuesday and user picture

Change picture for the user and tuesday

The default pictures are lackluster and needed to be changed.

Went on google to find a popular meme and save them as the pictures.

Darren-Tung committed:
Create Ui.png

Based on the requirement, Ui.png is created under the docs folder.

This will be used for product website and the user guide.

Darren-Tung committed:
Update README.md

The README.md has not been updated in a long time.

There is a need for a User Guide.

Updated the README for the User Guide as per requirement.

Darren-Tung committed:
Update the JavaDoc for Command Classes

There is a need to update the JavaDoc

This is due to the requirement of the iP

Updated JavaDoc for these classes:
- AddCommand
- DeleteCommand
- EditCommand
- ExitCommand
- FindCommand
- HelpCommand
- ListCommand
- MarkCommand
